Use inaccessible content attributes as old browser fallback.

peterwilsoncc opened the pull request:
Adds fallback `content` properties for browsers that do not support alt text.

In each case that the content property includes alt text, it is preceded by a version without alt text for browsers that do not support the feature. For example:

```css
.thing:before {
  content: "\f157";
  content: "\f157" / '';
}
```

Trac ticket: https://core.trac.wordpress.org/ticket/64350

peterwilsoncc committed:
Use inaccessible content attributes as old browser fallback.

peterwilsoncc committed:
Close unclosed string.

sabernhardt commented on src/wp-admin/css/customize-widgets.css:
```diff
@@ -308,73 +311,135 @@ body.adding-widget #customize-preview {
+#available-widgets [class*="categor"] .widget-title:before {
+	content:"\f318";
+	content:"\f318" / '';
+	top: -4px; }
```
Minor edits:
- The closing `}` (curly brace/bracket) is not on its own line in this ruleset.
- Starting with the 'dashicons-category' ruleset, `content` rules lack a space after the colon in `customize-widgets.css`.

```suggestion
	content: "\f318";
	content: "\f318" / '';
	top: -4px;
}
```

peterwilsoncc committed:
Place closing brace on own line.

Co-authored-by: Stephen A. Bernhardt <sabernhardt@yahoo.com>

peterwilsoncc committed:
Add spaces before property values.

Co-authored-by: Stephen A. Bernhardt <sabernhardt@yahoo.com>

peterwilsoncc commented:
Thanks @sabernhardt, I've fixed each of those issues. I did a search for `content:['"]` and unable to find any instances in other files that I've modified.

sabernhardt reviewed (approved):
I was able to reproduce the missing icons with Firefox 115 plus Safari 13.1.2 on my old Mac mini.

## Testing Results

✅ Issue resolved with patch.

Patch tested: https://github.com/WordPress/wordpress-develop/pull/10636.diff (as of [commit 9585212](https://github.com/WordPress/wordpress-develop/commit/95852123da6e2e2931d4a9c64b58da91a2ee6534))

### Checkboxes

Firefox 115 did not show the checked state of the checkbox with 6.9 styles, including the Remember Me checkbox on the login screen.
<img width="328" height="254" alt="6.9 login does not show that checkbox is checked" src="https://github.com/user-attachments/assets/0b250987-e9e4-49f7-a5d6-579cbd86ab80" />

The check icon appears with the patch, though I needed to change the version (to '6.9-alpha' in `version.php`) to make the login screen update its concatenated CSS without debug.
<img width="1067" height="698" alt="with patch, login shows check in checkbox" src="https://github.com/user-attachments/assets/f6f02340-4479-4911-b788-c9ee4dbbca37" />

The Classic Editor Add Post screen shows many missing icons with 6.9, and the Screen Options in this image are all checked without the icon (or browser default appearance).
<img width="1165" height="876" alt="6.9 does not show that all Screen Options in Classic Editor are checked" src="https://github.com/user-attachments/assets/5b34ec86-6b51-44a2-aacf-0caecfab0cbc" />

The patch restored icons for the checkboxes (without needing to change the version) plus icons in the admin toolbar, TinyMCE editor buttons, Publish and Format meta boxes, etc.
<img width="1058" height="876" alt="All Screen Options checked, with patch" src="https://github.com/user-attachments/assets/e153aaeb-ccef-4a60-86b9-893ce6868008" />

### Admin Toolbar

| WordPress 6.9 | Compiled CSS from Patch |
| ---- | ---- |
| With the admin dashboard, the menu toggle is the only Core icon shown on narrow screens<br><img width="751" height="471" alt="toolbar and admin dashboard on narrow screen, before patch" src="https://github.com/user-attachments/assets/edfdb750-6b1a-4548-a20e-3260ccde029e" /> | Core icons show in toolbar links (plus the meta box controls, At a Glance widget, and the Screen Options and Help buttons)<br><img width="745" height="471" alt="toolbar and admin dashboard on narrow screen, with patch" src="https://github.com/user-attachments/assets/b4b0bc16-e384-4d49-a27b-aec911a21a47" /> |
| Toolbar on the front end, on narrow screens<br><img width="709" height="495" alt="toolbar on the front, narrow screen, before patch" src="https://github.com/user-attachments/assets/8eeb8c8a-ff05-4bc1-9389-3abb45c2d4da" /> | Core icons display, including the default logo and arrows in the My Sites menu<br><img width="760" height="495" alt="toolbar on the front, narrow screen, with patch" src="https://github.com/user-attachments/assets/e3b9a67f-1108-467c-8d59-967e7007a590" /> |
| Front end, on wider screen<br><img width="1164" height="330" alt="toolbar showing submenu on the front, before patch" src="https://github.com/user-attachments/assets/e74004b6-7504-451e-b2f0-1b479c8cd8bf" /> | Core icons display again (the Edit Page link is there because this is a static front page)<br><img width="1058" height="330" alt="toolbar showing submenu on the front, before patch" src="https://github.com/user-attachments/assets/3e8603f7-5787-4f72-aae8-935c6f46ae3c" /> |
| Modern and Blue color schemes<br><img width="955" height="32" alt="toolbar in Modern color scheme, before patch" src="https://github.com/user-attachments/assets/fd75d10a-7c8d-4a31-8ea2-970f7834aa33" /><br><img width="955" height="32" alt="toolbar in Blue color scheme, before patch" src="https://github.com/user-attachments/assets/46dce3cb-f634-4083-b7ef-1e02d53ff348" /> | Icons in Modern and Blue color schemes<br><img width="1042" height="32" alt="toolbar in Modern color scheme, with patch" src="https://github.com/user-attachments/assets/cc4e9c32-86ee-4fab-8080-d97383fb7c6c" /><br><img width="1042" height="32" alt="toolbar in Blue color scheme, with patch" src="https://github.com/user-attachments/assets/2a4de702-947f-45f4-8dbb-ecfbc3b7629f" /> |

The Test Reports plugin showed its toolbar icon because it has its own CSS in `test-reports-admin-bar.css`:
```
#wp-admin-bar-test-reports .ab-icon::before {
	content: "\f451";
}
```

<img width="1165" height="359" alt="toolbar with Test Reports plugin, before patch" src="https://github.com/user-attachments/assets/23beb1d9-a127-4315-bcd1-8b0612d1a439" />

Similarly, the MonsterInsights analytics plugin showed its toolbar icon because it uses a `span` with `dashicons` classes.
`<span class="ab-icon dashicons-before dashicons-chart-bar"></span>`

<img width="1165" height="387" alt="toolbar with MonsterInsights analytics plugin, before patch" src="https://github.com/user-attachments/assets/39c4e54c-6a04-4e42-9d4e-ba9001728082" />

Both plugins' icons still show when the Core icons are restored.

<img width="955" height="32" alt="toolbar with Test Reports and MonsterInsights plugins, with patch" src="https://github.com/user-attachments/assets/cca75d94-c1e8-4618-b0f8-259ce2b6f241" />

### Additional Firefox screenshots

| WordPress 6.9 | Compiled CSS from Patch |
| ---- | ---- |
| Plugin details box does not show the close button (or the stars for rating)<br><img width="981" height="859" alt="Plugin details modal, before patch" src="https://github.com/user-attachments/assets/f8cccd53-bd9b-48fb-890f-f65481b96960" /> | The close button appears, with focus outline, and the rating shows stars<br><img width="1060" height="859" alt="Plugin details modal, with patch" src="https://github.com/user-attachments/assets/ced70b21-fb8e-4b5c-8728-f568a7c95dc6" /> |
| Categories list without sorting icons<br><img width="510" height="258" alt="Categories list table, before patch" src="https://github.com/user-attachments/assets/25a942f1-1dfa-4920-916e-378cefc44f8e" /> | List table with icons<br><img width="568" height="258" alt="Categories list table, with patch" src="https://github.com/user-attachments/assets/a3040f4e-c20d-48d6-bdfe-908108e41cda" /> |
| Media Library without icons for view switcher buttons<br><img width="523" height="116" alt="Media library view switcher buttons, in Grid mode, before patch" src="https://github.com/user-attachments/assets/ce5f1ca8-5ac5-43a9-8b94-74eccb43f781" /> | View switcher buttons show that this is in Grid mode<br><img width="523" height="116" alt="Media library view switcher buttons, in Grid mode, with patch" src="https://github.com/user-attachments/assets/82d20913-a835-478e-96da-faa0441bf965" /> |
| Image editor (from Grid mode)<br><img width="985" height="313" alt="Image editor dialog, before patch" src="https://github.com/user-attachments/assets/c2798d7e-8223-42cb-b372-00b9dfcf070c" /> | Icons appear for previous/next navigation, close button, and the editing controls<br><img width="1059" height="313" alt="Image editor dialog, with patch" src="https://github.com/user-attachments/assets/83182929-3e16-4a71-aac5-c989dac7bb10" /> |
| Theme Details has incorrect icon for previous/next navigation (r26141 assigned `dashicons-no` class to all 3 buttons)<br><img width="971" height="876" alt="Theme overlay, before patch" src="https://github.com/user-attachments/assets/e3480130-9742-464f-b193-6bf93d5c6d05" /> | Navigation and Close buttons use the correct icons<br><img width="969" height="500" alt="Theme overlay, with patch" src="https://github.com/user-attachments/assets/011e671e-6943-4935-ac4c-b43e730bbfd7" /> |
| Add Themes screen<br><img width="963" height="409" alt="Add Theme screen, before patch" src="https://github.com/user-attachments/assets/dcb6b3d0-3dcd-4160-8a44-93800fc5dd3d" /> | Icons restored for Feature Filter button and installed theme checkmark<br><img width="955" height="409" alt="Add Theme screen, with patch" src="https://github.com/user-attachments/assets/1e5f4b4e-218f-427b-86cc-6c7cbc280c6e" /> |
| The Customizer lacks the close and back arrow icons, and the Menus panel is missing icons for its add, delete, and toggle buttons<br><img width="674" height="640" alt="Customizer primary menu, before patch" src="https://github.com/user-attachments/assets/c74285af-1ba7-4a41-be6b-4ca391800bd0" /> | Icons appear: Customizer close and back arrow, red X for delete buttons, rotated Add Items toggle button icon, plus icons for Pages and the Add button in the extra panel<br><img width="674" height="640" alt="Customizer primary menu, with patch" src="https://github.com/user-attachments/assets/e69528e8-900c-4c72-b6e8-bc7de9ed0d2d" /> |
| Customizer Widgets: 'Add a Widget' panel also lacks icons for each widget<br><img width="674" height="875" alt="Customizer widgets, before patch" src="https://github.com/user-attachments/assets/dce3a05d-921a-4f8b-a644-70b860770148" /> | Available widgets have their icons<br><img width="674" height="875" alt="Customizer widgets, with patch" src="https://github.com/user-attachments/assets/4f8a561c-92a4-4830-90c2-9097d128dd0a" /><br>(I also captured [more Customizer widgets, with patch](https://github.com/user-attachments/assets/36ae7e52-dc60-46c1-afb6-6b055c15e48e), including the custom Twenty Eleven Ephemera widget) |
| Additional CSS panel information lacks icon for external links<br><img width="299" height="580" alt="Customizer Additional CSS, before patch" src="https://github.com/user-attachments/assets/e383203c-1df6-4f78-b80f-339e1a0cda2f" /> | External links show icon<br><img width="299" height="580" alt="Customizer Additional CSS, with patch" src="https://github.com/user-attachments/assets/a5a912d2-f686-4816-a592-5ce39ee9da91" /> |
| Customizer saving options lack the trash icon for Discard changes button<br><img width="674" height="450" alt="Customizer save options, before patch" src="https://github.com/user-attachments/assets/c9990796-3b8d-417a-9f5f-a8c4de4dc21e" /> | Discard changes button shows trash icon<br><img width="674" height="450" alt="Customizer save options, with patch" src="https://github.com/user-attachments/assets/b9ccf2a7-987b-45ac-bb66-8c17e851dcfc" /> |

### Safari screenshots

I had trouble with JavaScript in Safari 13.1.2, so I could not test as much there, but a few items displayed differently than in Firefox.

| WordPress 6.9 | Compiled CSS from Patch |
| ---- | ---- |
| Classic Editor: Add Post screen, without icons <br><img width="1043" height="612" alt="New post screen with Classic Editor, using Safari 13, before patch" src="https://github.com/user-attachments/assets/0f73c777-64df-48b1-8ecb-791ec04ba475" /> | Add Post screen, with icons<br><img width="1043" height="612" alt="New post screen with Classic Editor, using Safari 13, with patch" src="https://github.com/user-attachments/assets/2c679c47-154d-408c-8830-7e899606ac67" /> |
| Thickbox modal shows the close button's focus outline in front of the top area, but not the icon<br><img width="1046" height="349" alt="Plugin details modal, using Safari 13, before patch" src="https://github.com/user-attachments/assets/a25d3384-e62b-47ea-b75e-5248f7a75a2d" /> | Plugin details showing the icon inside the close button<br><img width="1046" height="349" alt="Plugin details modal, using Safari 13, with patch" src="https://github.com/user-attachments/assets/ca7d6331-577b-4109-a4d1-a5ce17779304" /> |

### Compiled CSS

- For most stylesheets, the empty alternative text string retains the single quotes. The minified versions remove the spaces before and after the slash.
- The stylesheets compiled for the color schemes use double quotes and remove the spaces surrounding the slash in all four versions (LTR or RTL, unminified or minified).

`media.css` and `media-rtl.css`
```
.imgedit-redo:before {
	content: "\f172";
	content: "\f172" / '';
}
```
`media.min.css` and `media-rtl.min.css`
```
.imgedit-redo:before{content:"\f172";content:"\f172"/''}`
```
`blue/colors.css` and `blue/colors-rtl.css`
```
input[type=checkbox]:checked::before {
  content: url("data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2020%2020%27%3E%3Cpath%20d%3D%27M14.83%204.89l1.34.94-5.81%208.38H9.02L5.78%209.67l1.34-1.25%202.57%202.4z%27%20fill%3D%27%237e8993%27%2F%3E%3C%2Fsvg%3E");
  content: url("data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2020%2020%27%3E%3Cpath%20d%3D%27M14.83%204.89l1.34.94-5.81%208.38H9.02L5.78%209.67l1.34-1.25%202.57%202.4z%27%20fill%3D%27%237e8993%27%2F%3E%3C%2Fsvg%3E")/"";
}
```
`blue/colors.min.css` and `blue/colors-rtl.min.css`
```
input[type=checkbox]:checked::before{content:url("data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2020%2020%27%3E%3Cpath%20d%3D%27M14.83%204.89l1.34.94-5.81%208.38H9.02L5.78%209.67l1.34-1.25%202.57%202.4z%27%20fill%3D%27%237e8993%27%2F%3E%3C%2Fsvg%3E");content:url("data:image/svg+xml;utf8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2020%2020%27%3E%3Cpath%20d%3D%27M14.83%204.89l1.34.94-5.81%208.38H9.02L5.78%209.67l1.34-1.25%202.57%202.4z%27%20fill%3D%27%237e8993%27%2F%3E%3C%2Fsvg%3E")/""}
```

### Environment
- WordPress: 6.9 before patch (after applying the patch to trunk-61390 locally, I used `npm run dev` and then replaced the entire `css` directories in `wp-admin` and `wp-includes` plus `thickbox.css` in my staging site with the compiled files)
- PHP: 8.2.29
- Server: Apache
- Database: mysqli (Server: 8.0.43-34 / Client: mysqlnd 8.2.29)
- Browsers:
   * Firefox 115.0.3 for the before and after screenshots, and I also checked 115.31.0esr
   * Safari 13.1.2 13609.3.5.1.5
- OS: macOS High Sierra 10.13.6 (17G14042)
- Theme: Twenty Eleven 5.0
- MU Plugins: None activated
- Plugins:
    * Classic Editor 1.6.7
    * Classic Widgets 0.3
    * Test Reports 1.2.1
- Debug: off (`WP_DEBUG` and `SCRIPT_DEBUG` false)
- Multisite

joedolson reviewed (approved):
LGTM!
